File: leoz-ui/src/main/java/org/deku/leo2/fx/modules/DebugController.java
package org.deku.leo2.fx.modules;

import javafx.fxml.FXML;
import javafx.fxml.Initializable;
import javafx.scene.control.CheckBox;
import javafx.scene.control.TextField;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;
import org.deku.leo2.Main;
import org.deku.leo2.Settings;
import org.deku.leo2.bridge.LeoBridge;
import org.deku.leo2.bridge.Message;
import org.deku.leo2.fx.ModuleController;

import java.net.URL;
import java.util.Date;
import java.util.ResourceBundle;

/**
 * Created by masc on 27.09.14.
 */
public class DebugController extends ModuleController implements Initializable {
    Log mLog = LogFactory.getLog(DebugController.class);

    @FXML
    private TextField mLeoBridgeMessageTextField;
    @FXML
    private CheckBox mUiAnimationsEnabled;

    @Override
    public String getTitle() {
        return "Debug";
    }

    @Override
    public void initialize(URL location, ResourceBundle resources) {
        mUiAnimationsEnabled.setSelected(Settings.instance().isAnimationsEnabled());
        mUiAnimationsEnabled.selectedProperty().addListener((o, ov, nv) -> {
            Settings.instance().setAnimationsEnabled(nv);
        });
    }

    public void onLeoBridgeSend() {
        try {
            LeoBridge.instance().sendMessage(new Message(mLeoBridgeMessageTextField.getText()));
        } catch(Exception e) {
            mLog.error(e.getMessage(), e);
        }
    }

    public void onLeoBridgeSendComplexMessage() {
        Message msg = new Message();
        msg.put("text", "hey");
        msg.put("int", 123);
        msg.put("float", 34.5);
        msg.put("date", new Date());
        try {
            LeoBridge.instance().sendMessage(msg);
        } catch(Exception e) {
            mLog.error(e.getMessage(), e);
        }
    }

    public void onDepotSelect() {
        Main.instance().getMainController().showModule(Main.instance().getMainController().getDepotMaintenanceModule());
        Main.instance().getMainController().getDepotMaintenanceModule().selectDepot(800);
    }
}

File: leoz-ui/src/main/java/org/deku/leoz/fx/modules/DebugController.kt
package org.deku.leoz.fx.modules

import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.control.CheckBox
import javafx.scene.control.TextField
import org.deku.leoz.Main
import org.deku.leoz.Settings
import org.deku.leoz.bridge.LeoBridge
import org.deku.leoz.bridge.Message
import org.deku.leoz.fx.ModuleController
import org.slf4j.LoggerFactory
import java.net.URL
import java.util.*

/**
 * Created by masc on 27.09.14.
 */
class DebugController : ModuleController(), Initializable {
    internal var log = LoggerFactory.getLogger(DebugController::class.java)

    @FXML
    private lateinit var fxLeoBridgeMessageTextField: TextField
    @FXML
    private lateinit var fxUiAnimationsEnabled: CheckBox

    override val title: String
        get() = "Debug"

    override fun initialize(location: URL, resources: ResourceBundle) {
        fxUiAnimationsEnabled.isSelected = Settings.instance().isAnimationsEnabled
        fxUiAnimationsEnabled.selectedProperty().addListener { o, ov, nv -> Settings.instance().isAnimationsEnabled = nv }
    }

    fun onLeoBridgeSend() {
        try {
            LeoBridge.instance().sendMessage(Message(fxLeoBridgeMessageTextField.text))
        } catch (e: Exception) {
            log.error(e.message, e)
        }

    }

    fun onLeoBridgeSendComplexMessage() {
        val msg = Message()
        msg.put("text", "hey")
        msg.put("int", 123)
        msg.put("float", 34.5)
        msg.put("date", Date())
        try {
            LeoBridge.instance().sendMessage(msg)
        } catch (e: Exception) {
            log.error(e.message, e)
        }

    }

    fun onDepotSelect() {
        Main.instance().mainController.showModule(Main.instance().mainController.depotMaintenanceController)
        Main.instance().mainController.depotMaintenanceController.selectDepot(800)
    }
}
